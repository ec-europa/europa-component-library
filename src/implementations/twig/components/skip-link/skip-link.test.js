import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import specs from '@ecl/specs-component-skip-link/demo/data';

expect.extend(toHaveNoViolations);

describe('Skip Link', () => {
  const template = '@ecl/skip-link/skip-link.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(specs)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(specs, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(specs, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(renderTwigFileAsHtml(template, specs, true))
      ).toHaveNoViolations();
    });
  });
});
