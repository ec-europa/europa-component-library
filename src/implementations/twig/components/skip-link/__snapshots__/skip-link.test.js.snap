// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skip Link Default renders correctly 1`] = `
<jest>
  <a
    class="ecl-link ecl-link--primary ecl-skip-link"
    href="#top"
  >
    Skip to main content
  </a>
</jest>
`;

exports[`Skip Link Default renders correctly with extra attributes 1`] = `
<jest>
  <a
    class="ecl-link ecl-link--primary ecl-skip-link"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
    href="#top"
  >
    Skip to main content
  </a>
</jest>
`;

exports[`Skip Link Default renders correctly with extra class names 1`] = `
<jest>
  <a
    class="ecl-link ecl-link--primary ecl-skip-link custom-class custom-class--test"
    href="#top"
  >
    Skip to main content
  </a>
</jest>
`;
